src/sync/incomingSyncUtils.js

```diff
@@ -306,9 +306,32 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
   let internalRecord;
   switch (recordType) {
     case 'IndicatorAttribute': {
+      internalRecord = {
+        id: record.ID,
+        indicator: database.getOrCreate('ProgramIndicator', record.indicator_ID),
+        description: record.description,
+        code: record.code,
+        index: parseNumber(record.index),
+        isRequired: parseBoolean(record.is_required),
+        valueType: record.data_type && record.data_type.value,
+        valueDefault: record.data_type && record.data_type.default,
+        axis: record.axis,
+        isActive: parseBoolean(record.is_active),
+      };
+      database.update(recordType, internalRecord);
       break;
     }
     case 'IndicatorValue': {
+      const recordValue = record.value && JSON.parse(record.value);
+      internalRecord = {
+        id: record.ID,
+        facilityId: record.facility_ID,
+        period: database.getOrCreate('Period', record.period_ID),
+        column: database.getOrCreate('IndicatorAttribute', record.column_ID),
+        row: database.getOrCreate('IndicatorAttribute', record.row_ID),
```

Comment:
hm, same as above these `IndicatorAttribute` records require a `ProgramIndicator` which isn't going to be created if they can't be getted, so I believe it'll throw. We need to propogate the creation of the `ProgramIndicator` on top of this. There's no helpers or mechanisms for this at the moment for this

Reply:
Fixed by d926c4c6.